File: android/src/NativePortListener.java
// SPDX-License-Identifier: GPL-2.0-or-later
// Copyright The XCSoar Project

package org.xcsoar;

/**
 * An #PortListener implementation that passes method calls to native
 * code.
 */
final class NativePortListener implements PortListener {
  /**
   * A native pointer.
   */
  private final long ptr;

  NativePortListener(long _ptr) {
    ptr = _ptr;
  }

  @Override public native void portStateChanged();
  @Override public native void portError(String msg);
}

File: src/Apple/NativePortListener.cpp
// SPDX-License-Identifier: GPL-2.0-or-later
// Copyright The XCSoar Project

#include "NativePortListener.hpp"
#include "Device/Port/Listener.hpp"

namespace NativePortListener {

void
Initialise()
{
  // No-op
}

void
Deinitialise()
{
  // No-op
}

PortListener *
Create(PortListener &listener)
{
  return &listener;
}

} // namespace NativePortListener
